app/models/user.rb

```diff
@@ -32,9 +38,18 @@ def friend_requests
   end
 
   def confirm_friend(user)
-    friendship = inverse_friendships.find { |i| i.user == user }
+    friendship = inverse_friendships.find { |i| i.user.id == user }
     friendship.confirmed = true
-    friendship.save
+    begin
+      ActiveRecord::Base.transaction do
```

Comment:
I'm not sure why you need this. Do you really need to catch any error here?

Reply:
Just to make sure that if there were any errors to occur that both transactions would be rolled back and that way I didn't have a loose solo friendship in the database. I removed this feature, it was mainly there as insurance.